packages/medusa/src/services/order-edit.ts

```diff
@@ -210,7 +226,13 @@ export default class OrderEditService extends TransactionBaseService {
           "shipping_methods.tax_lines",
         ],
       })
-    const { items } = await this.computeLineItems(orderEditId)
+    let { items } = await this.computeLineItems(orderEditId)
+
+    // For the purpose of the computation, we have to map the item id to the new item id instead of the original one for the correspondence of the tax lines and adjustments
```

Comment:
@srindom would like your opinion on that. We have decided that the items collection should be in a way where the item id correspond to the original one. Therefore the tax lines and adjustments are not allocated the right way. In the mean time for the purpose of the computation i have re mapped locally the id to get the proper expected computation to happen. From your opinion should we keep the item id as the one on the order or should it be the cloned one?

Reply:
Hmm good point - from an API consumer perspective, I think it would be okay to simply "hide" the `item_id` on the tax_lines since the object nesting would signify the relationship, so the id is not needed really; could even be extended to also hiding the `tax_line.id` itself so you can't find the inconsistency if you look up the tax line in the DB. 

That said, I really don't have strong opinions on this. I don't think we will experience issues with either approach, including simply ignoring the inconsistency; so will leave it up to you to decide how you want to proceed :)